### PR TITLE
TDD

### PR DESCRIPTION
**Pull request by PerryBaran**

### CODE REVIEW

File: src/kata1.fizzBuzz.js [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+  }
+
+  if (number % 5 === 0) {
+    return "Buzz";
+  }
+
+  if (number % 3 === 0) {
+    return "Fizz";
+  }
+  return number
```
Comment (PerryBaran) [inline src/kata1.fizzBuzz.js:13]: you call `number % 5 === 0` and `number % 3 === 0` multiple times throughout this function. You could make the function dryer by assigning these values to constants and them calling them instead, i.e.
```
const isDivisibleBy5 = number % 5 === 0;
const isDivisibleBy3 = number % 3 === 0;

if (isDivisibleBy5 && isDivisibleBy3) { ...
```

File: src/kata2.booleanToWord.js [hunk 2]
```diff
@@ -1,3 +1,11 @@
-const booleanToWord = (boolean) => {};
+const booleanToWord = (boolean) => {
+  if (boolean === true) {
+    return "yes";
+  }
```
Comment (PerryBaran) [inline src/kata2.booleanToWord.js:4]: when calling an `if` condition on a boolean value, or any value that you just want to check is true, you can instead just call `if (boolean)`

File: src/kata2.booleanToWord.js [hunk 3]
```diff
@@ -1,3 +1,11 @@
...
+
+  if (boolean === false) {
+    return "No";
+  }
```
Comment (PerryBaran) [inline src/kata2.booleanToWord.js:8]: because you've already checked if the boolean is true prior, the boolean must be false if you have got to this part of the code, so instead you can just write `return "No"`, i.e.
```
if (boolean) return "Yes";
return "No";
```

File: src/kata2.booleanToWord.js [hunk 4]
```diff
@@ -1,3 +1,11 @@
...
+};
```
Comment (PerryBaran) [inline src/kata2.booleanToWord.js:9]: This function can be further simplified by using the ternary operator, i.e.
```
return boolean ? "Yes" : "No";
```

File: src/kata4.humanCatDogYears.js [hunk 5]
```diff
@@ -1,3 +1,11 @@
-const humanCatDogYears = (number) => {};
+const humanCatDogYears = (number) => {
+  const ageTwo = 24;
+  const extraCatYears = (number - 2) * 4;
+  const catYears = ageTwo + extraCatYears;
+  const extraDogYears = (number - 2) * 5;
+  const dogYears = ageTwo + extraDogYears;
+  const yearsArray = [number, catYears, dogYears];
+  return yearsArray;
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:8]: Although this function works for all tests provided, if you test for `number = 0` and `number = 1` you get the wrong result

File: test/kata4.humanCatDogYears.test.js [hunk 6]
```diff
@@ -1,3 +1,8 @@
 const { humanCatDogYears } = require("../src");
 
-// Look Ma, no handlebars!!!
+describe("humanCatDogYears", () => {
+  test("returns array of human, cat and dog years when passed human years", () => {
+    expect(humanCatDogYears(15)).toEqual([15, 76, 89]);
+    expect(humanCatDogYears(3)).toEqual([3, 28, 29]);
+  });
+});
```
Comment (PerryBaran) [inline test/kata4.humanCatDogYears.test.js:8]: You should add more tests that test for the edge cases, e.g. human years = 1 and human years = 0.

when human years = 0 the output should be `[0, 0, 0]` but instead you return `[0, 16, 14]`

when human years = 1 the output should be `[1, 15, 15]`, but instead you return `[`, 20, 19]`

File: src/kata5.reachDestination.js [hunk 7]
```diff
@@ -1,3 +1,7 @@
-const reachDestination = (distance, speed) => {};
+const reachDestination = (distance, speed) => {
+  const journeyTime = distance / speed;
+  const roundedUp = Math.round(journeyTime * 2) / 2;
```
Comment (PerryBaran) [inline src/kata5.reachDestination.js:3]: I would give this constant a better name, as you aren't rounding up but instead are rounding to the nearest .5. For instance, if `journeyTime = 1.2`, `roundedUp = 1`, which is rounded down

File: src/kata5.reachDestination.js [hunk 8]
```diff
@@ -1,3 +1,7 @@
...
+  return `I will be there in ${roundedUp} hours`;
```
Comment (PerryBaran) [inline src/kata5.reachDestination.js:4]: if `roundedUp = 1` you should return `"... 1 hour"`, however at the moment you return `"... 1 hours"`

File: test/kata5.reachDestination.test.js [hunk 9]
```diff
@@ -1,5 +1,8 @@
 const { reachDestination } = require("../src");
 
 describe("reachDestination", () => {
-  test("returns string with estimated time of arrival", () => {});
+  test("returns string with estimated time of arrival", () => {
+    expect(reachDestination(66, 10)).toEqual("I will be there in 6.5 hours");
+    expect(reachDestination(44, 10)).toEqual("I will be there in 4.5 hours");
+  });
```
Comment (PerryBaran) [inline test/kata5.reachDestination.test.js:7]: I would add another test for the edge case when the rounded journey time = 1, as you'd expect a slightly different outcome to be grammatically correct, as mentioned previously.

File: src/kata6.joinNames.js [hunk 10]
```diff
@@ -1,3 +1,7 @@
-const joinNames = (namesObj) => {};
+const joinNames = (namesObj) => {
+  const namesArr = namesObj.map(({ name }) => name);
+  const lastName = namesArr.pop();
+  return `${namesArr.join(", ")} & ${lastName}`;
+};
```
Comment (PerryBaran) [inline src/kata6.joinNames.js:5]: This function works for the test provided, but if you pass an array with only 1 name you get the wrong output, i.e. `joinNames([{ name: "Bart" }])` return `" & Bart"`

File: test/kata6.joinNames.test.js [hunk 11]
```diff
@@ -1,5 +1,35 @@
...
+          name: "Tifa",
+        },
+        {
+          name: "Zack",
+        },
+        {
+          name: "Aeris",
+        },
+      ])
+    ).toEqual("Cloud, Tifa, Zack & Aeris");
```
Comment (PerryBaran) [inline test/kata6.joinNames.test.js:33]: You should add more tests for edge cases, i.e. when the array of namesObj has length 1 or 2.

When the array length is 1 you currently get the wrong output.

When the array length is 2 you get the expected output, but you should still have test for this edge case just to confirm this and to ensure that the functionality doesn't break if you were to refactor it.

File: src/kata7.getEmployerRole.js [hunk 12]
```diff
@@ -1,3 +1,8 @@
-const getEmployerRole = (employeeName, employees) => {};
+const getEmployerRole = (employeeName, employees) => {
+    const workerName = employees.find(
```
Comment (PerryBaran) [inline src/kata7.getEmployerRole.js:2]: I would rename this constant, as it isn't equal to the workers name but is instead equal to to the full worker object, you are just finding the worker by their name.

**Review by PerryBaran — COMMENTED** [review]
Lots of good stuff here, here's the summarised feedback:

- You need to add more test cases for edge cases. There are a few functions that work for the tests you've added but fail if you add tests for the edge cases.
- Some functions can be refactored to be DRY-er or to have more descriptive/relevant variable names
- Something I haven't mentioned elsewhere is that there is a lot of linting issues throughout your project. If you have a red squiggly line and/or the file name shows up as red in your IDE make sure to fix these. If you hover your mouse over the red squiggly line it will tell you what the issue is and then you can select the button `Quick Fix... (Ctrl + .)` to fix the issue.
